### PR TITLE
Add functorch M1 shard

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #85565
* #85564

functorch should have a test wherever regular PyTorch gets tested. This
PR adds an M1 shard to test functorch.

Test Plan:
- wait for CI

### CODE REVIEW

**Comment by zou3519** [comment]
not ready for review yet

File: .github/workflows/_mac-test.yml [hunk 1]
```diff
@@ -77,6 +77,7 @@ jobs:
       PR_BODY: ${{ github.event.pull_request.body }}
       PYTORCH_RETRY_TEST_CASES: 1
       PYTORCH_OVERRIDE_FLAKY_SIGNAL: 1
+      CONTINUE_THROUGH_ERROR: true
```
Comment (zou3519) [inline .github/workflows/_mac-test.yml:80]: I'll delete this before merging...

**Comment by zou3519** [comment]
Should be ready for review

**Comment by zou3519** [comment]
Some notes for the reviewers:
- the functorch M1 shard takes 1 hour. Not sure if this impacts TTS.
- artifacts: AFICT, none of the macos jobs, including the one added from this PR, have artifacts (https://hud.pytorch.org/pr/85565), though I might not be looking in the right place

**Review by malfet — APPROVED** [review]
LGTM, though it does not seem to be adding the shard, is it?

**Comment by zou3519** [comment]
> LGTM, though it does not seem to be adding the shard, is it?

Oops, I accidentally reverted that change. Will put it back, thanks for pointing it out.

**Review by huydhn — APPROVED** [review]
LGTM!

**Comment by huydhn** [comment]
>     * the functorch M1 shard takes 1 hour. Not sure if this impacts TTS.

Looking at the job https://github.com/pytorch/pytorch/actions/runs/3131300487/jobs/5082787746, it looks like pytest-shard works well for this as you can see 3 processes running in parallel there. That was done by https://github.com/pytorch/pytorch/pull/85321 and it was 2+ hours before.  1 hour is good :)

>     * artifacts: AFICT, none of the macos jobs, including the one added from this PR, have artifacts (https://hud.pytorch.org/pr/85565), though I might not be looking in the right place

This is a known connectivity issue between M1 runner and S3 IIRC cc @seemethere

**Comment by huydhn** [comment]
On the other hand, you can still get the artifacts from GItHub PR itself https://github.com/pytorch/pytorch/actions/runs/3131300487 (you need to scroll down or just click on the artifacts number there)

Screen Shot 2022-09-27 at 14 34 09

**Comment by zou3519** [comment]
Thanks for pointing out the parallelism and the artifacts, @huydhn

**Pull request closed without merging**